spec/factories/order_item_factory.rb

```diff
@@ -3,7 +3,7 @@
     order
     item
     sequence(:quantity) { |n| ("#{n}".to_i+1)*2 }
-    sequence(:price) { |n| ("#{n}".to_i+1*1.5) }
+    sequence(:price) { |n| (item && item.price || "#{n}".to_i+1*1.5) }
```

Comment:
this line says, if `order_item` is given and item, then set the price of `order_item` to that items current price else give it a value based on the logic that was already there. This enables us to give `order_item` an item in our tests and it's price is set correctly.